iptables/iptables.go

```diff
@@ -39,8 +40,43 @@ func (e *Error) Error() string {
 	return fmt.Sprintf("exit status %v: %v", e.ExitStatus(), e.msg)
 }
 
+// In earlier versions of iptables, the xtables lock was implemented via a unix socket, but now flock is used
+// via this lockfile (http://git.netfilter.org/iptables/commit/?id=aa562a660d1555b13cffbac1e744033e91f82707)
+const xtablesLockFile = "/run/xtables.lock"
+
+type xtablesFileMutex struct {
+	file *os.File
+}
+
+func (fmutex *xtablesFileMutex) Lock() (err error) {
+	fmutex = &xtablesFileMutex{}
```

Comment:
This is broken - you need to replace the dereferenced struct: http://play.golang.org/p/v-5GaSkF5G

But rather than playing around with all of this nil setting/checking, I'd rather just take an approach closer to that in rkt's [lock package](https://github.com/coreos/rkt/blob/master/pkg/lock/dir.go) (although much simpler), where locks are instantiated with a constructor and then the lock/unlock functions are very simple syscall wrappers (that can assume their receiver has been appropriately instantiated)